independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ComponentsProviderGenerator.java

```diff
@@ -282,9 +282,9 @@ private void addBean(MethodCreator getComponents, ResultHandle beanIdToBeanSuppl
                 beanInstance);
 
         // Create a Supplier that will return the bean instance at runtime.
-        ResultHandle beanInstanceSupplier = getComponents.newInstance(MethodDescriptors.FIXED_VALUE_SUPPLIER_CONSTRUCTOR,
-                beanInstance);
-        beanToResultSupplierHandle.put(bean, beanInstanceSupplier);
+
+        beanToResultSupplierHandle.put(bean,
+                () -> getComponents.newInstance(MethodDescriptors.FIXED_VALUE_SUPPLIER_CONSTRUCTOR, beanInstance));
```

Comment:
Shouldn't this cache the value in case .get() is called multiple times?

Reply:
I don't think it *needs* to, but sure, it could be an optimisation. It will reduce the number of instances of `FixedValueSupplier` at runtime.